### PR TITLE
Auth: Update docs on use PKCE by default

### PR DESCRIPTION
**Pull request by arukiidou**
**What is this feature?**

- Update docs for #68095
  - Issue: #68073

**Why do we need this feature?**

- PKCE is going to be a required part of OAuth 2.1 which means that in the near future it'll be something that'll be required by authorization servers.
- And some OAuth provider supports PKCE.
  - see #22122
    - also see #39948

**Who is this feature for?**

- Users who use an Oauth provider

**Special notes for your reviewer**:

- Somehow an index was added to azuread, but I have not modified it. I will revert if not needed.
- I assume #68095 will be included in 10.1

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 1]
```diff
@@ -191,6 +201,18 @@ The `allowed_domains` option limits access to users who belong to specific domai
...
+### PKCE
+
+> Available in Grafana v8.3 and later versions.
+
+IETF's [RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636)
+introduces "proof key for code exchange" (PKCE) which introduces
+additional protection against some forms of authorization code
+interception attacks. PKCE will be required in [OAuth 2.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-03).
+
+> **Note:** Grafana v10.1 and newer use `use_pkce = true` by default,
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:213]:
```suggestion
> **Note:** Grafana v10.1 and newer use `use_pkce = true` by default.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 2]
```diff
@@ -191,6 +201,18 @@ The `allowed_domains` option limits access to users who belong to specific domai
...
+> You can disable PKCE in Grafana by setting `use_pkce` to `false` in the`[auth.azuread]` section.
+
 ### Configure automatic login
 
 Set `auto_login` option to true to attempt login automatically, skipping the login screen.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:218]:
```suggestion
If you want to automatically log in without having to go through the login screen, enable the `auto_login` option.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 3]
```diff
@@ -142,8 +143,27 @@ role_attribute_path = is_admin && 'Admin' || 'Viewer'
 role_attribute_strict = true
 allow_assign_grafana_admin = false
 tls_skip_verify_insecure = false
+use_pkce = true
 ```
 
+### PKCE
+
+> Available in Grafana v8.3 and later versions.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:151]:
```suggestion
```

Comment (chri2547) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:151]: We have versioned docs and don't need notes like this.

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 4]
```diff
@@ -142,8 +143,27 @@ role_attribute_path = is_admin && 'Admin' || 'Viewer'
...
+
+IETF's [RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636)
+introduces "proof key for code exchange" (PKCE) which introduces
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:154]:
```suggestion
introduces "proof key for code exchange" (PKCE) which provides
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 5]
```diff
@@ -142,8 +143,27 @@ role_attribute_path = is_admin && 'Admin' || 'Viewer'
...
+additional protection against some forms of authorization code
+interception attacks. PKCE will be required in [OAuth 2.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-03).
+
+> **Note:** Grafana v10.1 and newer use `use_pkce = true` by default,
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:158]:
```suggestion
> **Note:** Grafana v10.1 and newer uses `use_pkce = true` by default,
```

Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:158]:
```suggestion
```

Comment (chri2547) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:158]: Not necessary to add a Note about a version that hasn't come out yet.

File: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md [hunk 6]
```diff
@@ -58,6 +59,18 @@ automatically signed up.
 You may specify a domain to be passed as `hd` query parameter accepted by Google's
 OAuth 2.0 authentication API. Refer to Google's OAuth [documentation](https://developers.google.com/identity/openid-connect/openid-connect#hd-param).
 
+### PKCE
+
+> Available in Grafana v8.3 and later versions.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:64]:
```suggestion
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md [hunk 7]
```diff
@@ -51,8 +51,21 @@ api_url = https://<tenant-id>.okta.com/oauth2/v1/userinfo
 allowed_domains =
 allowed_groups =
 role_attribute_path =
+use_pkce = true
 ```
 
+### PKCE
+
+> Available in Grafana v8.3 and later versions.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md:59]:
```suggestion
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md [hunk 8]
```diff
@@ -51,8 +51,21 @@ api_url = https://<tenant-id>.okta.com/oauth2/v1/userinfo
...
+
+IETF's [RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636)
+introduces "proof key for code exchange" (PKCE) which introduces
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md:62]:
```suggestion
introduces "proof key for code exchange" (PKCE) which provides
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md (hunk 2 above)
Comment (arukiidou) [reply docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:218]: This was not included in my changes.
Are you sure you want to make this change? (If you👍 then I will do.)

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md [hunk 9]
```diff
@@ -191,6 +201,18 @@ The `allowed_domains` option limits access to users who belong to specific domai
 allowed_domains = mycompany.com mycompany.org
 ```
 
+### PKCE
+
+> Available in Grafana v8.3 and later versions.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:206]:
```suggestion
```

**Review by chri2547 — CHANGES_REQUESTED** [review]
Added some suggestions. @linoman or @IevaVasiljeva , would you be able to take a look at this PR, please? Thank you.

**Review by IevaVasiljeva — APPROVED** [review]
The doc changes look good to me. Thanks a lot for your contribution!

**Comment by arukiidou** [comment]
@chri2547 Resolved, except the question.
What about two questions?

File: docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md (hunk 2 above)
Comment (chri2547) [reply docs/sources/setup-grafana/configure-security/configure-authentication/azuread/index.md:215]: @arukiidou , my apologies, I didn't see this one. I adjusted the suggestion slightly. Feel free to commit it.

**Review by chri2547 — APPROVED** [review]
Thanks for the contribution!

**Comment by chri2547** [comment]
> @chri2547 Resolved, except the question. What about two questions?

Hi @arukiidou , I only saw one question for me, just updated the suggestion. If you are ok with my wording, please go ahead and accept and then I'll merge this in. Thanks so much for the contribution!

**Comment by arukiidou** [comment]
> > @chri2547 Resolved, except the question. What about two questions?
> 
> Hi @arukiidou , I only saw one question for me, just updated the suggestion. If you are ok with my wording, please go ahead and accept and then I'll merge this in. Thanks so much for the contribution!

@chri2547 
I appreciate your kindness!

What about this question?
https://github.com/grafana/grafana/pull/68638#discussion_r1200505602

**Comment by chri2547** [comment]
@arukiidou , I'm seeing that conflicts need to be resolved. Is that something you can do? I'm unfamiliar with which change to accept.

**Comment by arukiidou** [comment]
> @arukiidou , I'm seeing that conflicts need to be resolved. Is that something you can do? I'm unfamiliar with which change to accept.

- Yes, I did it. 
  - Resolve conflicts brought from https://github.com/grafana/grafana/pull/68228

**Pull request merged**